scheduler/scheduler.cpp

```diff
@@ -1816,6 +1823,86 @@ static void trigger_exit(int signum)
     signal(signum, trigger_exit);
 }
 
+static void handle_scheduler_announce(char* buf, const char* netname, int schedbuflen, bool persistent_clients, struct sockaddr_in broad_addr)
+{
+    /* Another scheduler is announcing it's running, disconnect daemons if it has a better version
+       or the same version but was started earlier. */
+    if (!persistent_clients){
+        uint64_t tmp_time;
+        memcpy(&tmp_time, buf + 4, sizeof(uint64_t));
+        time_t other_time = tmp_time;
+        int other_scheduler_protocol = buf[3];
+        string recv_netname;
+        if (other_scheduler_protocol >= 36)
+        {
+            buf[schedbuflen] = '\0';
```

Comment:
We should not modify buf here - future protocols might add more data to the message that we would be clobbering.